source/new-image-handler/src/processor/image/watermark.ts

```diff
@@ -527,8 +527,7 @@ export class WatermarkAction extends BaseImageAction {
 function withNormalSize(metadata: sharp.Metadata): sharp.Metadata {
   const o = Object.assign({}, metadata);
   if ((metadata.orientation || 0) >= 5) {
-    o.width = o.height;
-    o.height = o.width;
+    [o.width, o.height] = [o.height, o.width];
```

Comment:
use this 
  if ((metadata.orientation || 0) >= 5) {
    o.width = metadata.height;
    o.height = metadata.width;
  }

Reply:
https://stackoverflow.com/a/25910841